src/charm.py

```diff
@@ -940,569 +472,497 @@ def _handle_status_update_config(self, event: HookEvent) -> None:
-        elif not identity_id and not email:
-            event.fail("One of identity-id and email must be provided.")
-            return None
-
-        if identity_id:
-            try:
-                uuid.UUID(identity_id)
-            except ValueError:
-                event.fail("The identity-id must be a valid UUID.")
-                return None
-
-        if email:
-            identity = self.kratos.get_identity_from_email(email)
-            if not identity:
-                event.fail("Couldn't retrieve identity_id from email.")
-                return None
-            identity_id = identity["id"]
-
-        return identity_id
-
     def _on_delete_identity_action(self, event: ActionEvent) -> None:
-        if not self._kratos_service_is_running:
+        if not self._workload_service.is_running:
             event.fail("Service is not ready. Please re-run the action when the charm is active")
             return
 
-        if not (identity_id := self._get_identity_id(event)):
-            return
-
-        event.log("Deleting the identity.")
-        try:
-            self.kratos.delete_identity(identity_id=identity_id)
-        except Error as e:
-            event.fail(f"Something went wrong when trying to run the command: {e}")
-            return
+        identity_id = event.params.get("identity-id")
+        with HTTPClient(base_url=f"http://127.0.0.1:{KRATOS_ADMIN_PORT}") as client:
+            try:
+                client.delete_identity(identity_id)
+            except IdentityNotExistsError:
+                event.fail(f"Identity {identity_id} does not exist")
+                return
+            except ClientRequestError:
+                event.fail(f"Failed to delete the identity for {identity_id}")
+                return
 
-        event.log(f"Successfully deleted the identity: {identity_id}.")
-        event.set_results({"identity-id": identity_id})
+        event.log(f"Successfully deleted the identity: {identity_id}")
+        event.set_results({"id": identity_id})
 
     def _on_reset_password_action(self, event: ActionEvent) -> None:
-        if not self._kratos_service_is_running:
+        if not self._workload_service.is_running:
             event.fail("Service is not ready. Please re-run the action when the charm is active")
             return
 
-        if not (identity_id := self._get_identity_id(event)):
-            return
-
-        if secret_id := event.params.get("password-secret-id"):
+        password = None
+        password_secret_id = event.params.get("password-secret-id")
+        if password_secret_id:
             try:
-                juju_secret = self.model.get_secret(id=secret_id)
+                juju_secret = self.model.get_secret(id=password_secret_id)
                 password = juju_secret.get_content().get("password")
             except SecretNotFoundError:
-                event.fail("Secret not found")
+                event.fail("Juju secret is not found")
```

Comment:
nit: the previous message sounds more natural
```suggestion
                event.fail("Juju secret not found")
```